numpy/core/arrayprint.py

```diff
@@ -759,17 +768,21 @@ def __init__(self, data, precision, floatmode, suppress_small, sign=False,
 
         self._legacy = kwarg.get('legacy', False)
         if self._legacy == '1.13':
-            sign = '-' if data.shape == () else ' '
+            # legacysign is a secret FloatingFormat option that adds a sign
+            # even in legacy mode, needed for imag of complex.
```

Comment:
How was this handled before?

Reply:
Right, in 1.13 the `sign` option took care of it. But now in 1.14 we have taken over and redefined `sign`, and since it is now ignored in legacy mode we need another way of forcing the '+'.

Reply:
I think this would also work:
```
if self._legacy == '1.13':
    # when not 0d, legacy does not support '-'
    if data.shape != () and sign == '-':
        sign = ' '
```

Reply:
True. I considered something similar but I was worried about people using `sign='+', legacy='1.13'` at the same time, and getting confused at non-legacy-like output since the `+` option didn't exist in 1.13.

But on second thought I'll go with your suggestion.